middleware.js

```diff
@@ -12,13 +14,33 @@ module.exports.isLoggedIn = (req, res, next) => {
 };
 
 module.exports.fetchRandomGameData = async (req, res, next) => {
-  const randGameId = Math.floor(Math.random() * 1000 + 1);
-  const results = await axios.get(
-    `https://api.rawg.io/api/games/${randGameId}?key=${process.env.RAWG_KEY}`
-  );
-  req.gameData = results.data;
+  let response = null;
+  let retries = 1;
+  let success = false;
+  const maxRetries = 5;
 
-  next();
+  while (retries <= maxRetries && !success) {
+    try {
+      const randGameId = Math.floor(Math.random() * 500000 + 1);
+      response = await axios.get(
+        `https://api.rawg.io/api/games/${randGameId}?key=${process.env.RAWG_KEY}`
```

Comment:
I would want to see the API url to declared in a constant

Reply:
declared it in the file, might move it elsewhere later depending on how the code will develop